lib/python/CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+# Python extension library for mayaUsd library bindings. Pixar's TF_WRAP_MODULE
+# macro expects the library name to be prefixed with _, but adds an underscore
+# to the package name when creating the Python extension module initialization
+# function (see Pixar's pyModule.h).
+if(IS_WINDOWS AND MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG)
+    # On Windows when compiling with debug python the library must be named with _d.
+    set(PYTHON_TARGET_NAME _${PROJECT_NAME}_d)
+else()
+    set(PYTHON_TARGET_NAME _${PROJECT_NAME})
+endif()
+
+add_library(${PYTHON_TARGET_NAME} SHARED)
+
+# -----------------------------------------------------------------------------
+# sources
+# -----------------------------------------------------------------------------
+target_sources(${PYTHON_TARGET_NAME} 
+    PRIVATE
+        module.cpp
+        moduleDeps.cpp
```

Comment:
I think there's a subtle distinction in purpose between these two files that's not represented here.

`module.cpp` is definitely intended to get built into the Python module library object (`lib/python/mayaUsd/lib/_mayaUsd.so` on Linux), but `moduleDeps.cpp` is intended to get built into the main library (`lib/libmayaUsd.so` on Linux).

For pxr libraries, `module.cpp` is identified as a `PYMODULE_CPPFILES`:
https://github.com/PixarAnimationStudios/USD/blob/40c813421425c47aa84f4572b4798fbb3430984e/cmake/macros/Public.cmake#L304
Whereas `moduleDeps.cpp` is identified as a `PYTHON_CPPFILES`:
https://github.com/PixarAnimationStudios/USD/blob/40c813421425c47aa84f4572b4798fbb3430984e/cmake/macros/Public.cmake#L219

The upshot is that I think `moduleDeps.cpp` need to go up a level and be included in the main lib/CMakeLists.txt

Reply:
Ugh I didn't know. Will fix it in the next commit. Thank you for the links.

Reply:
please see 12f14b651